### DISCUSSION TITLE
Update Pro from previous Pro and..?

### DISCUSSION THREAD

**Discussion by clive7391** [1N3/Sn1per#322] [Q&A] [answered]
I have used Pro and all subsequent add-ons since v5 as a secondary testing tool. There has always been an upgrade option and I haven't seen that.

Is there an upgrade (loyalty) upgrade to 9?
Why is there a limit on workspace IPs? I scan tens of thousands of IPs in a workspace and a limit is an issue.

**Comment by 1N3** [comment, marked as answer]
Please see here for our upgrade special ([https://xerosecurity.com/wordpress/sn1per-professional-v9-0-whats-new/](https://xerosecurity.com/wordpress/sn1per-professional-9-0-customer-loyalty-offer/)). 

This was posted at the bottom of our latest blog post here: [https://xerosecurity.com/wordpress/sn1per-professional-v9-0-whats-new/](https://xerosecurity.com/wordpress/sn1per-professional-v9-0-whats-new/)
We will probably send an email notification since some may not have noticed this. 

The workspace limits were put in place for peformance reasons as Pro v9.0 is more intensive than previous versions due to all the added features. 

We feel that 50,000 hosts should be more than sufficient for most usage based on the current pricing model (we've seen similar commercial tools charge many times this for way less...). To clarify, you can scan an unlimited amount of IP's - it's the web UI that will have limits on how much is displayed. If you are scanning different subnets or IP ranges, it would be best to break these up into more manageable workspaces if possible. 

Hope that helps and thanks again for the support.

**Reply by clive7391** [reply]
So to clarify, if you had command line scanner before, you will get it free if you buy 9?

**Reply by 1N3** [reply]
Essentially, the offer if you upgrade to Pro v9.0 is to get the new Command Execution Add-on v2.0 free with your order. The command execution add-on allows you to execute scans via the web UI which ties into the command line version of Sn1per.

**Reply by clive7391** [reply]
Last question, will ALL the previous add-ons (minus CE) work with 9?
Port scanner
Threat Intel
Brute
Fuzzer
Nessus

So ultimately it will be elite?

**Reply by 1N3** [reply]
Yes, all add-ons from 8.0 aside from CE v1.0 will work with 9.0. To upgrade to 9.0 and upgrade the add-ons, you just have to run the activation.sh script over top of your existing 8.0 installation and it will upgrade the add-ons automatically to work with 9.0.

**Reply by clive7391** [reply]
Thanks and Purchased
